beetsplug/limit.py

```diff
@@ -0,0 +1,95 @@
+# This file is part of beets.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+from beets.dbcore import FieldQuery
+from beets.plugins import BeetsPlugin
+from beets.ui import Subcommand, decargs, print_
+from collections import deque
+from itertools import islice
+
+
+def lslimit(lib, opts, args):
+    """Query command with head/tail"""
+    
+    head = opts.head
+    tail = opts.tail
+
+    if head and tail:
+        raise RuntimeError("Only use one of --head and --tail")
+    
+    query = decargs(args)
+    if opts.album:
+        objs = lib.albums(query)
+    else:
+        objs = lib.items(query)
+    
+    if head:
+        objs = islice(objs, head)
+    elif tail:
+        objs = deque(objs, tail)
+
+    for obj in objs:
+        print_(format(obj))
+
+
+lslimit_cmd = Subcommand(
+    "lslimit", 
+    help="query with optional head or tail"
+)
+
+lslimit_cmd.parser.add_option(
+    '--head', 
+    action='store', 
+    type="int",
+    default=None
+)
+
+lslimit_cmd.parser.add_option(
+    '--tail',
+    action='store', 
+    type="int",
+    default=None
+)
+
+lslimit_cmd.parser.add_all_common_options()
+lslimit_cmd.func = lslimit
+
+
+class LsLimitPlugin(BeetsPlugin):
+    def commands(self):
+        return [lslimit_cmd]
+
+
+class HeadPlugin(BeetsPlugin):
```

Comment:
It's probably not necessary to group this into two separate `BeetsPlugin` objects. The typical way to do this (in most plugins) is to just put everything into one class—i.e., this latter plugin could have a `commands` method as well.